### DISCUSSION TITLE
Initialization state?

### DISCUSSION THREAD

**Discussion by jonayreyes** [RomanIakovlev/timeshape#86] [Q&A] [answered]
Hi! first of all, congratulations for this amazing piece of software!

I'd like to know if there's any way of knowing the internal state of the TimeZoneEngine, given the fact that the initialization for all the world takes approx 6 minutes on a Intel® Core™ i7-8650U CPU @ 1.90GHz × 8 with 16GB RAM.

I'd like to reject calls to the (only) .query methods, or at least return meaningful messages if the engine is not yet initialized

Any ideas?

Thanks in advance!

**Comment by RomanIakovlev** [comment, marked as answer]
Hi and thanks for your kind words! I'm glad you find Timeshape useful.

There's no mechanism that tracks the initialization right now, because in my experience Timeshape initializes quick enough. On my desktop computer with Core i5 6600 it takes less than 2 seconds, and in some other environments I've experienced similar initialization duration.

It's definitely possible to add such a mechanism, but I think it would be better to try and understand why the initialization takes so long for you. It's really hard to make a guess without knowing about your setup, but, as it frequently is the case with Java, the performance problems might be caused by misconfiguration of available heap space (or just lack of it) and garbage collectors, so I'd suggest taking a look there.

I'd be really curious to learn if you'd manage to find a root cause for this performance issue, so please keep me updated!

**Reply by jonayreyes** [reply]
Yes: given the fact that we're initializing it inside another hosting service, something was definitely blocking the spin up, because providing the TimeZoneEngine a dedicated thread solved the issue.

Thank you very much!
